src/lib.rs

```diff
@@ -423,79 +429,43 @@ impl Check {
             };
         }
 
-        let current_loader = self.get_rustdoc_generator(config, &self.current.source)?;
-        let baseline_loader = self.get_rustdoc_generator(config, &self.baseline.source)?;
-
-        // Create a report for each crate.
-        // We want to run all the checks, even if one returns `Err`.
-        let all_outcomes: Vec<anyhow::Result<(String, Option<CrateReport>)>> = match &self
-            .current
-            .source
-        {
+        let crates_to_check: Vec<CrateToCheck> = match &self.current.source {
```

Comment:
I find omitted lifetimes to be a bit of a footgun, so I'd love to acknowledge them even with just `'_`:
```suggestion
        let crates_to_check: Vec<CrateToCheck<'_>> = match &self.current.source {
```